frontend/src/views/WheresMyGene/components/HeatMap/components/YAxisChart/index.tsx

```diff
@@ -212,18 +212,19 @@ export default memo(function YAxisChart({
         isMarkerGenes &&
         yAxisInfoCoords.map((coord, i) => {
           const content = cellTypeMetadata[i];
+          const cellType = isMarkerGenes ? deserializeCellTypeMetadata(content) : null;
           return (
             <InfoButtonWrapper
               id={content}
               key={content}
               left={coord[0]}
               top={coord[1]}
               onClick={() => {
-                if (isMarkerGenes) {
-                  const cellType = deserializeCellTypeMetadata(content);
+                if (cellType) {
                   generateMarkerGenes(cellType, tissueID);
                 }
               }}
+              style={cellType && cellType.total_count < 25 ? {visibility: "hidden"} : {}}
```

Comment:
should we just return `null` instead of `visibility: "hidden"` 😄 ? `hidden` is good if we expect to unhide the DOM element after user interaction, but that doesn't seem to be the case here, so we can save some memory and CPU cycle on rendering unused elements 👌 !